### PR TITLE
Fill in the gaps right before idle mode onset

### PR DESCRIPTION
**Pull request by dgerstel**

### CODE REVIEW

File: pygt3x/reader.py [hunk 1]
```diff
@@ -116,6 +118,16 @@ def get_data(self, num_rows=None):
...
+                    try:
+                        last_second = acceleration[-1][0, 0]
+                    except IndexError:
+                        pass
+                    else:
+                        dt = evt.header.timestamp - last_second
+                        if dt >= 2:
+                            ts = pd.to_datetime(evt.header.timestamp, unit="s")
+                            logging.debug(f"Missed {dt}s before {ts}")
+
```
[the remark this answers is not in the document]
Comment (dgerstel) [reply pygt3x/reader.py]: Two checks added:
- each event is required to have sample frequency number of records (`_validate_payload` method); otherwise a ValueError is thrown
- at the end of `get_data()` assertions check if each second appears only sample frequency times in the whole dataset (the previous check doesn't guarantee that this one should pass, because there may be duplicated seconds each with sample frequency records)

Two fixes added:
- fill the gaps right before idle sleep mode is on
- in case the current event timestamp occurs before an already filled-in idle sleep mode period ends, then update the accelerometer with the current event data

Tested on 5 ALS participants.

TODO: 
- Fix 1 failling test `test_read_v1` from `test_formats.py` (fails because 30 Hz data happens to result in 1 record per second). Investigating...
- Add unit tests for the above-mentioned edge cases. I'll have a look if it's possible to simulate these edge cases without the need to provide heavy gt3x files.

**Comment by dgerstel** [comment]
@rouzbeh 
Ready for further review / merge
Replaced exceptions with warnings in case of missing data in payloads

**Review by rouzbeh — APPROVED** [review]
